firebase-firestore/proguard.txt

```diff
@@ -16,3 +16,7 @@
 # Okhttp warnings.
 -dontwarn okio.**
 -dontwarn com.google.j2objc.annotations.**
+
+# Kotlin warnings:
+-dontwarn org.jetbrains.kotlinx:kotlinx-serialization-core:1.3.2
+-dontwarn org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlinVersion
```

Comment:
I have added the Kotlin dep to the proguard file

Reply:
But this will fail the smoke test:
```
Calling mockable JAR artifact transform to create file: /root/.gradle/caches/transforms-3/b8b8649dc08110bfac85/root/.gradle/caches/transforms-3/3d83f850dfc61654f267737bad7ad40a/transformed/proguard.txt:22:33: D8: Expected char '-' at /root/.gradle/caches/transforms-3/3d83f850dfc61654f267737bad7ad40a/transformed/proguard.txt:21:32
-dontwarn org.jetbrains.kotlinx:kotlinx-serialization-core:1.3.2
                               ^

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':transformClassesAndResourcesWithR8ForRelease'.
> com.android.tools.r8.CompilationFailedException: Compilation failed to complete
```

Reply:
After some research, I think what I need to do are:
1. replace all the `:` with `.`;
2. replace all the `-` with `_`
To get R8 works together with ProGuard.txt